enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,475 @@
+the upstream CRD source implementation to allow for other record types to be created via the DNSRecord CRD on OpenShift, such as `TXT` or `AAAA` records.
+
+__Note:__ Any API(s) introduced by this enhancement will follow the Kubernetes API
+[deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/). Example API groups are also subject to change.
+
+### User Stories
+
+#### Story 1
+
+As a developer, I need the ability to manage DNS records for OpenShift routes without manually creating
+and updating DNS records by hand.
+
+#### Story 2
+
+As a cluster administrator, I need the ability use a single operator to manage external DNS requirements for all
+platform components.
+
+#### Story 3
+
+As a security-mindful cluster administrator, I cannot in good conscious let the Ingress Operator create wildcard DNS records
+for cluster Ingress. I need to have the ability to manage fully-qualified DNS records for cluster workloads with Routes.
+
+#### Story 4
+
+As a cluster administrator, I want ExternalDNS to create DNS records for all of my LoadBalancer services in Route53,
+in addition to creating DNS records for all of my OpenShift routes in BlueCat DNS, so that I can easily manage my multi-provider
+DNS requirements.
+
+To satisfy this, a cluster administrator could create 2 different ExternalDNS Operator Custom Resources, one for each cloud provider.
+
+#### Story 5
+
+As a cluster administrator, I want the ExternalDNS Operator to deploy one ExternalDNS instances for each of my DNS zones, with each deployment
+being configured to use a different set of credentials, since I do not want to reuse common credentials for both zones due to security and auditing concerns.
+
+### Implementation Details
+
+When using the ExternalDNS Route Source on OpenShift, ExternalDNS will create CNAME record for a
+Route's host field pointing to the Route status' `routerCanonicalHostname` field.
+An in-flight [PR for the Ingress Operator](https://github.com/openshift/cluster-ingress-operator/pull/610)
+will enhance the `routerCanonicalHostname` field so that the field has a unique domain that falls under
+the Ingress Controller's given domain (to ensure compatibility with existing wildcard DNS records for Ingress).
+
+As an alternative to using CNAME records, the ExternalDNS Route source implementation could point Route hostnames directly to
+LoadBalancer IP addresses or hostnames. This is not ideal since coordinating with administrator-managed load-balancers in this situation would be
+non-trivial. Using the `routerCanonicalHostname` field as a DNS middle-man (so to speak) simplifies ExternalDNS management.
+
+Note that the OpenShift [Route source implementation](https://github.com/kubernetes-sigs/external-dns/blob/master/source/ocproute.go)
+in ExternalDNS supports some of the common [ExternalDNS annotations](https://github.com/kubernetes-sigs/external-dns/blob/master/source/source.go),
+such as the ExternalDNS TTL annotation.
```

Comment:
If this note is an implementation detail, it would be good to elaborate on which annotations would be important.

Reply:
Added the Target annotation as another example.